push_notification/subscriber/services/subscriber_service.py

```diff
@@ -0,0 +1,48 @@
+from subscriber.models.subscriber import Subscriber
+
+
+class SubscriberService(object):
+
+    def __init__(self, subscriber, serializer):
+        self.subscriber = subscriber
+        self.serializer = serializer
+
+    def get_subscriber(self, pk=None):
+        "Return subscriber details by given id"
+        if pk is not None:
+            queryset = self.subscriber.objects.get(id=pk)
```

Comment:
We can handle the case where subscriber does not exist for the given id. In which case, the current code will raise an error. We can use a try block here. In case the object does not exist we can raise an exception `SubscriberDoesNotExist` which can be handled in the API service layer